cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/SpannerMetadataStorage.java

```
@@ -54,7 +316,19 @@ public String getName() {
 
   @Override
   public void dropIndex() throws IOException {
-    throw new IOException("NOT IMPLEMENTED");
+    synchronized (this) {
```

No need of synchronization here. You can just silently ignore exception if table does not exist.

---

Okk